src/Mysql/Delete.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * Copyright (C) 2016 Datto, Inc.
+ *
+ * This file is part of Cinnabari.
+ *
+ * Cinnabari is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * Cinnabari is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cinnabari. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Spencer Mortensen <smortensen@datto.com>
+ * @author Anthony Liu <aliu@datto.com>
+ * @license http://www.gnu.org/licenses/lgpl-3.0.html LGPL-3.0
+ * @copyright 2016 Datto, Inc.
+ */
+
+namespace Datto\Cinnabari\Mysql;
+
+use Datto\Cinnabari\Exception\AbstractException;
+use Datto\Cinnabari\Mysql\Expression\AbstractExpression;
+
+class Delete extends AbstractMysql
+{
+    public function getMysql()
+    {
+        if (!$this->isValid()) {
+            throw new AbstractException(
+                self::ERROR_INVALID_MYSQL,
+                array(),
+                "SQL delete queries must reference at least one table."
+            );
+        }
+
+        $mysql = "DELETE\n" .
+            $this->getTables() .
+            $this->getWhereClause() .
+            $this->getOrderByClause() .
+            $this->getLimitClause();
+
+        return rtrim($mysql, "\n");
+    }
+
+    public function setOrderBy($tableId, $column, $isAscending)
+    {
+        if (!self::isDefined($this->tables, $tableId)) {
+            $tableString = json_encode($tableId);
+            throw new AbstractException(
+                self::ERROR_BAD_TABLE_ID,
+                array(
+                    'tableId' => $tableId
+                ),
+                "unknown table id {$tableString}."
+            );
+        }
+
+        $table = $this->getTable($tableId);
+        $name = self::getAbsoluteExpression($table, $column);
+
+        $mysql = "ORDER BY {$name} ";
+
+        if ($isAscending) {
+            $mysql .= "ASC";
+        } else {
+            $mysql .= "DESC";
+        }
+
+        $this->orderBy = $mysql;
+    }
+
+    public function setLimit($tableId, AbstractExpression $length)
+    {
+        if (!self::isDefined($this->tables, $tableId)) {
+            return null;
+        }
+
+        $count = $length->getMysql();
+
+        $mysql = "{$count}";
+
+        $this->limit = $mysql;
+    }
+
+    protected function getTables()
```

Comment:
The "Delete" class is only required to support the simple (single-table) format of the MySQL [DELETE](http://dev.mysql.com/doc/refman/5.7/en/delete.html#idm139945013655872) syntax.